gprofiler/application_separators.py

```diff
@@ -0,0 +1,238 @@
+import configparser
+import os.path
+import re
+from abc import ABCMeta, abstractmethod
+from typing import List, Optional
+
+from granulate_utils.linux.ns import resolve_host_path
+from psutil import NoSuchProcess, Process
+
+from gprofiler.log import get_logger_adapter
+
+logger = get_logger_adapter(__name__)
+
+
+_PYTHON_BIN_RE = re.compile(r"^python([23](\.\d{1,2})?)?$")
+
+
+def _is_python_bin(bin_name: str):
+    return _PYTHON_BIN_RE.match(os.path.basename(bin_name)) is not None
+
+
+class ApplicationSeparator(metaclass=ABCMeta):
+    @abstractmethod
+    def is_supported(self, process: Process) -> bool:
+        pass
+
+    @abstractmethod
+    def get_application_name(self, process: Process) -> Optional[str]:
+        pass
+
+    @property
+    @abstractmethod
+    def app_prefix(self) -> str:
+        pass
+
+
+_IP_PORT_RE = re.compile(r"^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}(?:\:\d{2,5})?$")  # Matches against (ip(:port))
+_NON_AVAILABLE_ARG = str()
+
+
+def get_cli_arg_by_name(args: List[str], arg_name: str, check_for_equals_arg: bool = False) -> str:
+    if arg_name in args:
+        return args[args.index(arg_name) + 1]
+
+    if check_for_equals_arg:
+        for arg in args:
+            arg_name, _, arg_val = arg.rpartition("=")
+            if arg_val is not None:
+                return arg_val
+
+    return _NON_AVAILABLE_ARG
+
+
+def get_cli_arg_by_index(args: List[str], index: int) -> str:
+    try:
+        return args[index]
+    except KeyError:
+        return _NON_AVAILABLE_ARG
+
+
+def append_python_module_to_proc_wd(process: Process, module_name: str) -> str:
+    return f'{process.cwd().replace("/", ".").strip(".")}.{module_name}'
+
+
+class GunicornApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "gunicorn"
+
+    def is_supported(self, process: Process) -> bool:
+        # Either gunicorn (for example: /usr/local/bin/gunicorn) or python that runs gunicorn
+        # (For example: /usr/local/bin/python /usr/local/bin/gunicorn)
+        return "gunicorn" in get_cli_arg_by_index(process.cmdline(), 0) or "gunicorn" in get_cli_arg_by_index(
+            process.cmdline(), 0
+        )
```

Comment:
It's the same condition in both sides of the `or`

Reply:
Not addressed?

Reply:
Missed that

Reply:
Done, resolving.